src/Core/Grid/Definition/Factory/Security/Session/CustomerGridDefinitionFactory.php

```diff
@@ -0,0 +1,263 @@
+use PrestaShop\PrestaShop\Core\Grid\Definition\Factory\DeleteActionTrait;
+use PrestaShop\PrestaShop\Core\Grid\Filter\Filter;
+use PrestaShop\PrestaShop\Core\Grid\Filter\FilterCollection;
+use PrestaShop\PrestaShop\Core\Grid\Filter\FilterCollectionInterface;
+use PrestaShopBundle\Form\Admin\Type\DateRangeType;
+use PrestaShopBundle\Form\Admin\Type\SearchAndResetType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+
+/**
+ * Class CustomerGridDefinitionFactory responsible for creating Customer sessions grid definition.
+ */
+final class CustomerGridDefinitionFactory extends AbstractGridDefinitionFactory
+{
+    use BulkDeleteActionTrait;
+    use DeleteActionTrait;
+
+    /**
+     * @var string
+     */
+    public const GRID_ID = 'security_session_customer';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getId(): string
+    {
+        return static::GRID_ID;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getName(): string
+    {
+        return $this->trans('Customer sessions', [], 'Admin.Navigation.Menu');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getColumns(): ColumnCollectionInterface
+    {
+        return (new ColumnCollection())
+            ->add(
+                (new BulkActionColumn('bulk'))
+                    ->setOptions([
+                        'bulk_field' => 'id_customer_session',
+                    ])
+            )
+            ->add(
+                (new DataColumn('id_customer_session'))
+                    ->setName($this->trans('ID', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'id_customer_session',
+                    ])
+            )
+            ->add(
+                (new DataColumn('id_customer'))
+                    ->setName($this->trans('Customer ID', [], 'Admin.Advparameters.Feature'))
+                    ->setOptions([
+                        'field' => 'id_customer',
+                    ])
+            )
+            ->add(
+                (new DataColumn('firstname'))
+                    ->setName($this->trans('First name', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'firstname',
+                    ])
+            )
+            ->add(
+                (new DataColumn('lastname'))
+                    ->setName($this->trans('Last name', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'lastname',
+                    ])
+            )
+            ->add(
+                (new DataColumn('email'))
+                    ->setName($this->trans('Email', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'email',
+                    ])
+            )
+            ->add(
+                (new DataColumn('date_upd'))
+                    ->setName($this->trans('Updated at', [], 'Admin.Advparameters.Feature'))
```

Comment:
Will this strong be followed by a date or a time?

Updated **at** 4pm
Updated **on** March 17

Reply:
It's a date time format "Y-m-d H:i:s"

Reply:
Ok thanks. It should be "on" then. :)